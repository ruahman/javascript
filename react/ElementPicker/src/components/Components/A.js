import React from "react";

const A = ({ values }) => {
  const onClick = e => {
    e.preventDefault();
    window.open(values.href, "newBrowsingContext");
  };
  return (
    <a href={values.href} onClick={onClick}>
      {values.label}
    </a>
  );
};

export default A;
