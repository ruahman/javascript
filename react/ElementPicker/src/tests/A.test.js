import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import A from '../components/Components/A.js';

describe('A', () => {
  it('works ok', () => {
    const component = shallow((
      <A
        values={{
          href: 'http://www.google.com',
          label: 'Click me!',
        }}
      />
    ));

    expect(toJson(component)).toMatchSnapshot();
  });
});
