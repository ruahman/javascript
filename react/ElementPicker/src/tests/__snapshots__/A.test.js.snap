// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A works ok 1`] = `
<a
  href="http://www.google.com"
  onClick={[Function]}
>
  Click me!
</a>
`;
